### PR TITLE
feat: added compile lambda

### PR DESCRIPTION
**Pull request by taco-paco**

### CODE REVIEW

File: lambdas/src/compile.rs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    run, service_fn, Error as LambdaError, Request as LambdaRequest, Response as LambdaResponse,
+};
+use serde::Deserialize;
+use std::ops::Add;
+use tracing::{error, info};
+
+mod common;
+use crate::common::{errors::Error, utils::extract_request, Item, Status, BUCKET_NAME_DEFAULT};
+
+const QUEUE_URL_DEFAULT: &str = "https://sqs.ap-southeast-2.amazonaws.com/266735844848/zksync-sqs";
```
Comment (varex83) [inline lambdas/src/compile.rs:14]: I think we should set it via `env`

File: lambdas/src/compile.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    match result {
+        Ok(val) => val,
+        Err(SdkError::ServiceError(val)) => match val.err() {
+            PutItemError::ConditionalCheckFailedException(_) => {
+                error!("Recompilation attempt, id: {}", id);
+                let response = lambda_http::Response::builder()
+                    .status(400)
+                    .header("content-type", "text/html")
+                    .body("Recompilation attempt".into())
```
Comment (varex83) [inline lambdas/src/compile.rs]: Let's make "recompilation attemp" const too, as done for `QUEUE_URL_DEFAULT`?

**Review by varex83 — APPROVED** [review]
LGTM overall

File: lambdas/src/compile.rs (hunk 1 above)
Comment (taco-paco) [reply lambdas/src/compile.rs:14]: It's done [here](https://github.com/NethermindEth/zksync-remix-plugin/blob/a9d07097555df71776143b91811f95caf1211bef/lambdas/src/compile.rs#L149). that is just default for ease of use for now. Will add comment to remove

File: lambdas/src/compile.rs (hunk 2 above)
Comment (taco-paco) [reply lambdas/src/compile.rs]: done

File: lambdas/src/compile.rs (hunk 1 above)
Comment (taco-paco) [reply lambdas/src/compile.rs:14]: done

**Comment by taco-paco** [comment]
@varex83 will merge now. If any comments unresolved will free to add, I will address them in new branch

**Pull request merged**